packages/x-charts/src/BarChart/BarElement.tsx

```diff
@@ -98,6 +89,73 @@ export type BarElementProps = Omit<BarElementOwnerState, 'isFaded' | 'isHighligh
     slots?: BarElementSlots;
   };
 
+/**
+ * @ignore - This component is just here to remove the ownerState from slotProps before rendering.
+ */
+function AnimatedBarElement(props: BarProps) {
+  const { ownerState, ...other } = props;
+  return <animated.rect {...other} />;
+}
```

Comment:
Is this to prevent a breaking change? Should we move it to its own file to remove the `propTypes` below? 🙃

Reply:
> Is this to prevent a breaking change?

Not exactly to avoid a breaking change since I think we will struggle to remove it.

My issue is the `ownerState` that I would like to still provide to the `slots.bar` because it contains interesting information. But I can not pass it to the `animated.bar` because it would propagate this props as a SVG attribute.

> Should we move it to its own file to remove the propTypes below? 

Does moving it remove the proptypes? I just added the component to the list of component without proptypes, but if you've a better solution, would be happy to update it

Reply:
If this is not to prevent a breaking change we can simply destructure in the component itself, which would remove the need for this "in-between" component.

eg: `const {ownerState, ...otherProps} = useSlotProps({...})`

Reply:
But then the `ownerState` will not be available for the ``slots.bar` if the user provide one.

We would need to add `<Bar {...other} ownerState={slots.bar ? ownerState : undefined} />` to both:
- pass the ownerState to custom slot
- do not pass ownerState to SVG element

Reply:
That's why I asked if it was for a breaking change 😅 

I don't think the users should receive `ownerState` at all though...

Reply:
how would they customize the slots if the only thing they receive is the SVG attributes?

Reply:
ownerState is an abstraction for styled, all the props from ownerstate could be in the root props

Instead of `<Foo ownerState={{a:1, b:'bar'}} />` we would have `<Foo a={1} b={"bar"} />`

Reply:
I just move the problem from "`onwnerState` is not a SVG attribute" to "`isHighlighted``is not a SVG attribute". The advantage of `ownerState` is that all the properties to remove are in a single object you can quickly remove

Reply:
If I use material/styled for a while yeah, I understand what it does. 

But if I pick the library the first time? It feels hidden. I would rather just have everything in the root and destructure everything myself.

I think `ownerState` solves our issue, which is to have internal state that shouldn't be passed down. 

But it doesn't seen to solve an user issue, which is usually `what are all the available props and how I use them`. Since we are hiding props inside something with a generic name like `ownerState`.

Reply:
Ok, I get the point. But that's outside of this PR topic. It's a broader topic about "what do we pass to slots"

Reply:
Yup I agree